messaging/messaging.go

```diff
@@ -81,10 +81,23 @@ type Message struct {
 	Webpush      *WebpushConfig    `json:"webpush,omitempty"`
 	APNS         *APNSConfig       `json:"apns,omitempty"`
 	Token        string            `json:"token,omitempty"`
-	Topic        string            `json:"topic,omitempty"`
+	Topic        string            `json:"-"`
 	Condition    string            `json:"condition,omitempty"`
 }
 
+// MarshalJSON marshals a Message into JSON (for internal use only).
+func (m *Message) MarshalJSON() ([]byte, error) {
+	type messageInternal Message
```

Comment:
I'd add a comment explaining that you have to create a new type to prevent this function from infinitely recursing, assuming that's why it's here. It wasn't obvious.

Reply:
Done